week-2/mandatory/2-function-creation.js

```diff
@@ -0,0 +1,145 @@
+/*
+Write a function that:
+- takes an array of strings as input
+- removes any spaces in the beginning or end of the strings
+- removes any forward slashes (/) in the strings
+- makes the string all lowercase
+*/
+// ["/Daniel ", "irina ", " Gordon", "ashleigh "]
+function tidyUpString(strArr) {
+    for (let i = 0; i < strArr.length; i++) {
+        strArr[i] = strArr[i].trim().toLowerCase()
+        strArr[i] = strArr[i].replace("/", "")
+    }
+    return strArr
+}
+/*
+Complete the function to check if the variable `num` satisfies the following requirements:
+- is a number
+- is even
+- is less than or equal to 100
+Tip: use logical operators
+*/
+
+function validate(num) {
+    if (typeof num == "number" && num % 2 == 0 && num <= 100) {
+        return true
+    } else {
+        return false
+    }
+}
+/* 
+Write a function that removes an element from an array
+The function must:
+- NOT change the original array
+- return a new array with the item removed
+- remove the item at the specified index
+*/
+// let arr = [10, 293, 292, 176, 29]
+
+function remove(arr, index) {
+    arr.splice(index, 1)
+    return arr
```

Comment:
Nice, almost perfect! Have a look at the note in https://www.w3schools.com/jsref/jsref_splice.asp.

The tests didn't check the first requirement `The function must NOT change the original array`. Is it satisfied in this case? Try playing around a bit with this. Create a variable containing the array. Call your method on this variable and print it afterwards. Does the array stay unchanged? Could you think of a way to make sure the original array does not change?